src/validation.cpp

```diff
@@ -180,18 +180,14 @@ std::optional<std::vector<int>> CalculatePrevHeights(
     std::vector<int> prev_heights;
     prev_heights.resize(tx.vin.size());
     for (size_t i = 0; i < tx.vin.size(); ++i) {
-        const CTxIn& txin = tx.vin[i];
-        Coin coin;
-        if (!coins.GetCoin(txin.prevout, coin)) {
+        if (auto coin{coins.GetCoin(tx.vin[i].prevout)}) {
+            prev_heights[i] = coin->nHeight == MEMPOOL_HEIGHT
+                              ? tip.nHeight + 1 // Assume all mempool transaction confirm in the next block.
+                              : coin->nHeight;
+        } else {
             LogPrintf("ERROR: %s: Missing input %d in transaction \'%s\'\n", __func__, i, tx.GetHash().GetHex());
```

Comment:
@maflcko should I [use `LogInfo`](https://github.com/bitcoin/bitcoin/pull/29641/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98R186) here instead?

Reply:
Thanks, used `LogError` in [`90aa839` (#30849)](https://github.com/bitcoin/bitcoin/pull/30849/commits/90aa839456163a48e675f601ff6a05330aa5c5c6)

Reply:
> Also, changing the log severity is not a "refactor".

K, dropping this commit in that case